components/scroller/scroller.vue

```diff
@@ -0,0 +1,299 @@
+      const sizes = this.vscrollData.sizes;
+      const l = items.length;
+      for (let i = 0; i < l; i++) {
+        const item = items[i];
+        const id = simpleArray ? i : item[keyField];
+        let size = sizes[id];
+        if (typeof size === 'undefined' && !this.$_undefinedMap[id]) {
+          size = 0;
+        }
+        result.push({
+          item,
+          id,
+          size,
+        });
+      }
+      return result;
+    },
+  },
+
+  watch: {
+    items () {
+      this.forceUpdate();
+    },
+
+    simpleArray: {
+      handler (value) {
+        this.vscrollData.simpleArray = value;
+      },
+
+      immediate: true,
+    },
+
+    direction () {
+      this.forceUpdate(true);
+    },
+
+    itemsWithSize (next, prev) {
+      const scrollTop = this.$el.scrollTop;
+
+      // Calculate total diff between prev and next sizes
+      // over current scroll top. Then add it to scrollTop to
+      // avoid jumping the contents that the user is seeing.
+      let prevActiveTop = 0; let activeTop = 0;
+      const length = Math.min(next.length, prev.length);
+      for (let i = 0; i < length; i++) {
+        if (prevActiveTop >= scrollTop) {
+          break;
+        }
+        prevActiveTop += prev[i].size || this.minItemSize;
+        activeTop += next[i].size || this.minItemSize;
+      }
+      const offset = activeTop - prevActiveTop;
+
+      if (offset === 0) {
+        return;
+      }
+
+      this.$el.scrollTop += offset;
+    },
+  },
+
+  beforeCreate () {
+    this.$_updates = [];
+    this.$_undefinedSizes = 0;
+    this.$_undefinedMap = {};
+  },
+
+  activated () {
+    this.vscrollData.active = true;
+  },
+
+  deactivated () {
+    this.vscrollData.active = false;
+  },
+
+  methods: {
+    forceUpdate (clear = false) {
+      if (clear || this.simpleArray) {
+        this.vscrollData.sizes = {};
+      }
+    },
+
+    scrollToItem (index) {
+      const scroller = this.$refs.scroller;
+      if (scroller) scroller.scrollToItem(index);
+    },
+
+    getItemSize (item, index = undefined) {
+      const id = this.simpleArray ? (index != null ? index : this.items.indexOf(item)) : item[this.keyField];
+      return this.vscrollData.sizes[id] || 0;
+    },
+
+    scrollToBottom () {
+      if (this.$_scrollingToBottom) return;
+      this.$_scrollingToBottom = true;
+      const el = this.$el;
+      // Item is inserted to the DOM
+      this.$nextTick(() => {
+        el.scrollTop = el.scrollHeight + 5000;
+        // Item sizes are computed
+        const cb = () => {
+          el.scrollTop = el.scrollHeight + 5000;
+          requestAnimationFrame(() => {
+            el.scrollTop = el.scrollHeight + 5000;
+            if (this.$_undefinedSizes === 0) {
+              this.$_scrollingToBottom = false;
+            } else {
+              requestAnimationFrame(cb);
+            }
+          });
+        };
+        requestAnimationFrame(cb);
+      });
```

Comment:
No, this is from the library and it is used if it is necessary to have sth like a "Chat Mode".
I didn't tried it

Reply:
ye... mhmm let me see what I can do...
At this moment our scroller.vue its their dynamic scroller from the library, and our core-scroller its their recycle-scroller.

And Im just triggering one or the other if they set `dynamic` true

Reply:
> ye... mhmm let me see what I can do... At this moment our scroller.vue its their dynamic scroller from the library, and our core-scroller its their recycle-scroller.
> 
> And Im just triggering one or the other if they set `dynamic` true

The ideal way to do this would be to put the dynamic scroller code in it's own file within core as well. 

Logic to decide which is triggered or not based on whether dynamic = true would remain in the scroller.vue

Reply:
Agree, let me rethink this and I will separate `dynamic` to another component in `modules`